packages/vite/src/node/config.ts

```diff
@@ -967,40 +967,32 @@ async function bundleConfigFile(
       {
         name: 'externalize-deps',
         setup(build) {
+          const options: InternalResolveOptions = {
+            root: path.dirname(fileName),
+            isBuild: true,
+            isProduction: true,
+            isRequire: !isESM,
+            preferRelative: false,
+            tryIndex: true,
+            mainFields: DEFAULT_MAIN_FIELDS,
+            browserField: false,
+            conditions: [],
+            dedupe: [],
+            extensions: DEFAULT_EXTENSIONS,
+            preserveSymlinks: false
+          }
+
           build.onResolve({ filter: /.*/ }, ({ path: id, importer }) => {
             // externalize bare imports
-            if (id[0] !== '.' && !path.isAbsolute(id)) {
+            if (id[0] !== '.' && !isAbsolute(id)) {
+              const idFsPath = tryNodeResolve(id, importer, options, false)?.id
+              const idPath =
+                isESM && idFsPath ? pathToFileURL(idFsPath).href : idFsPath
```

Comment:
@antfu could Windows be complaining because `idFsPath` is now normalized? In the current code it isn't. I don't think this should affect though.